// ngx-horizontal-timeline/bundles/ngx-horizontal-timeline.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-horizontal-timeline', ['exports', '@angular/core', '@angular/common'], factory) :
    (factory((global['ngx-horizontal-timeline'] = {}),global.ng.core,global.ng.common));
}(this, (function (exports,i0,common) { 'use strict';

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxHorizontalTimelineService = /** @class */ (function () {
        function NgxHorizontalTimelineService() {
        }
        NgxHorizontalTimelineService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        NgxHorizontalTimelineService.ctorParameters = function () { return []; };
        /** @nocollapse */ NgxHorizontalTimelineService.ngInjectableDef = i0.defineInjectable({ factory: function NgxHorizontalTimelineService_Factory() { return new NgxHorizontalTimelineService(); }, token: NgxHorizontalTimelineService, providedIn: "root" });
        return NgxHorizontalTimelineService;
    }());

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m)
            return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length)
                    o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxHorizontalTimelineComponent = /** @class */ (function () {
        function NgxHorizontalTimelineComponent() {
        }
        /**
         * @return {?}
         */
        NgxHorizontalTimelineComponent.prototype.ngOnInit = /**
         * @return {?}
         */
            function () { };
        /**
         * @param {?} item
         * @return {?}
         */
        NgxHorizontalTimelineComponent.prototype.activeItem = /**
         * @param {?} item
         * @return {?}
         */
            function (item) {
                var e_1, _a;
                try {
                    for (var _b = __values(this.items), _c = _b.next(); !_c.done; _c = _b.next()) {
                        var i = _c.value;
                        i.active = undefined;
                    }
                }
                catch (e_1_1) {
                    e_1 = { error: e_1_1 };
                }
                finally {
                    try {
                        if (_c && !_c.done && (_a = _b.return))
                            _a.call(_b);
                    }
                    finally {
                        if (e_1)
                            throw e_1.error;
                    }
                }
                item.active = true;
                if (item.command) {
                    item.command();
                }
            };
        /**
         * @param {?} item
         * @return {?}
         */
        NgxHorizontalTimelineComponent.prototype.getColor = /**
         * @param {?} item
         * @return {?}
         */
            function (item) {
                if (!item.color) {
                    return null;
                }
                if (item.active) {
                    return '#' + item.color;
                }
                else {
                    return '#' + this.lighten(item.color, 50);
                }
            };
        /**
         * @param {?} col
         * @param {?} amt
         * @return {?}
         */
        NgxHorizontalTimelineComponent.prototype.lighten = /**
         * @param {?} col
         * @param {?} amt
         * @return {?}
         */
            function (col, amt) {
                col = parseInt(col, 16);
                return (
                // tslint:disable-next-line:no-bitwise
                ((col & 0x0000ff) + amt) | ((((col >> 8) & 0x00ff) + amt) << 8) | (((col >> 16) + amt) << 16)).toString(16);
            };
        NgxHorizontalTimelineComponent.decorators = [
            { type: i0.Component, args: [{
                        selector: 'ngx-htl-horizontal-timeline',
                        template: "<div id=\"timeline-wrap\">\n  <div class=\"navigation\">\n\n    <span *ngFor=\"let item of items\" [class]=\"item.styleClass? item.styleClass: ''\">\n      <div class=\"timeline-label\" *ngIf=\"item.label\">{{item.label}}</div>\n\n      <div (click)=\"activeItem(item)\" [style.background-color]=\"getColor(item)\"\n        title=\"{{item.title}}\"\n        [class]=\"\n          item.active\n            ? 'marker timeline-icon marker-active'\n            : 'marker timeline-icon'\">\n        <i [class]=\"item.icon\"></i>\n      </div>\n    </span>\n    <div class=\"timeline\"></div>\n  </div>\n</div>\n",
                        styles: [".navigation{display:flex;flex-flow:row wrap;justify-content:space-around;align-items:center}.timeline{height:1px;width:100%;position:absolute;background:#aabbc4}.timeline:before{background:0 0}.timeline-label{text-align:center;position:absolute;width:50px;bottom:55px;color:#555;font-family:\"Source Sans Pro\",\"Helvetica Neue\",Helvetica,Arial,sans-serif!important;font-weight:400}@media all and (max-width:500px){.navigation{flex-direction:row;justify-content:initial}.timeline{height:0}.marker{margin-top:25px}.timeline-title{position:relative;bottom:-14px}}#timeline-wrap{margin:4% 5%;position:relative}.marker{z-index:1;color:#fff;width:50px;height:50px;line-height:50px;font-size:1.4em;text-align:center;position:relative;background-color:#999;border-radius:50%}.marker:hover{-webkit-transform:scale(1.3);transform:scale(1.3);cursor:pointer;transition:.3s}.timeline-icon{background-color:#7e8aaf}.marker-active{-webkit-transform:scale(1.3);transform:scale(1.3);background-color:#3e4f88}"]
                    }] }
        ];
        /** @nocollapse */
        NgxHorizontalTimelineComponent.ctorParameters = function () { return []; };
        NgxHorizontalTimelineComponent.propDecorators = {
            items: [{ type: i0.Input }]
        };
        return NgxHorizontalTimelineComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgxHorizontalTimelineModule = /** @class */ (function () {
        function NgxHorizontalTimelineModule() {
        }
        NgxHorizontalTimelineModule.decorators = [
            { type: i0.NgModule, args: [{
                        declarations: [NgxHorizontalTimelineComponent],
                        imports: [
                            common.CommonModule
                        ],
                        exports: [NgxHorizontalTimelineComponent]
                    },] }
        ];
        return NgxHorizontalTimelineModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.NgxHorizontalTimelineService = NgxHorizontalTimelineService;
    exports.NgxHorizontalTimelineComponent = NgxHorizontalTimelineComponent;
    exports.NgxHorizontalTimelineModule = NgxHorizontalTimelineModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
