// angular-vertical-timeline/__ivy_ngcc__/dist/vertical-timeline.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VerticalTimelineComponent } from './vertical-timeline.component';
import { VerticalTimelineCardComponent } from './vertical-timeline-card.component';
import * as ɵngcc0 from '@angular/core';
var VerticalTimelineModule = /** @class */ (function () {
    function VerticalTimelineModule() {
    }
    VerticalTimelineModule.forRoot = function () {
        return {
            ngModule: VerticalTimelineModule,
        };
    };
VerticalTimelineModule.ɵfac = function VerticalTimelineModule_Factory(t) { return new (t || VerticalTimelineModule)(); };
VerticalTimelineModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: VerticalTimelineModule });
VerticalTimelineModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(VerticalTimelineModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    VerticalTimelineComponent,
                    VerticalTimelineCardComponent
                ],
                exports: [
                    VerticalTimelineComponent,
                    VerticalTimelineCardComponent
                ]
            }]
    }], function () { return []; }, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(VerticalTimelineModule, { declarations: function () { return [VerticalTimelineComponent,
        VerticalTimelineCardComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [VerticalTimelineComponent,
        VerticalTimelineCardComponent]; } }); })();
    return VerticalTimelineModule;
}());
export { VerticalTimelineModule };
