// angular-vertical-timeline/__ivy_ngcc__/dist/vertical-timeline.component.js
import { Component } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

var _c0 = ["*"];
var VerticalTimelineComponent = /** @class */ (function () {
    function VerticalTimelineComponent() {
    }
VerticalTimelineComponent.ɵfac = function VerticalTimelineComponent_Factory(t) { return new (t || VerticalTimelineComponent)(); };
VerticalTimelineComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: VerticalTimelineComponent, selectors: [["vertical-timeline"]], ngContentSelectors: _c0, decls: 3, vars: 0, consts: [[1, "timeline"]], template: function VerticalTimelineComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "section", 0);
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(VerticalTimelineComponent, [{
        type: Component,
        args: [{
                selector: 'vertical-timeline',
                template: "\n    <section class=\"timeline\">\n      <div>\n        <ng-content></ng-content>\n      </div>\n    </section>\n  "
            }]
    }], function () { return []; }, null); })();
    return VerticalTimelineComponent;
}());
export { VerticalTimelineComponent };
