// angular-vertical-timeline/__ivy_ngcc__/dist/vertical-timeline-card.component.js
import { Component, Input, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function VerticalTimelineCardComponent_p_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "p");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵpipe(2, "date");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("color", ctx_r0.textColor);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(2, 3, ctx_r0.dateValue, "HH:mm"));
} }
function VerticalTimelineCardComponent_p_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "p");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("color", ctx_r1.textColor);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.timeString);
} }
var _c0 = function (a0) { return { "disabled-element": a0 }; };
var _c1 = ["*"];
var VerticalTimelineCardComponent = /** @class */ (function () {
    function VerticalTimelineCardComponent() {
        this.disabled = false;
        this.isATimelineItem = false;
    }
    VerticalTimelineCardComponent.prototype.ngOnInit = function () {
        this.isATimelineItem = true; // set class `timeline-item` on host `<div>`
        if (this.dateValue === null || this.dateValue === undefined) {
            this.dateValue = new Date();
        }
        if (this.color === null || this.color === undefined) {
            this.color = '#3F51B5';
        }
        if (this.isLight(this.color)) {
            this.textColor = '#000000';
        }
        else {
            this.textColor = '#FFFFFF';
        }
    };
    VerticalTimelineCardComponent.prototype.isLight = function (hexColor) {
        var R = parseInt(hexColor.slice(1, 3), 16);
        var G = parseInt(hexColor.slice(3, 5), 16);
        var B = parseInt(hexColor.slice(5, 7), 16);
        var maxBrightness = this.calculateBrightness(255, 255, 255);
        var brightness = this.calculateBrightness(R, G, B);
        var pBrightness = brightness / maxBrightness;
        return pBrightness > 0.5;
    };
    // HSP rule sqrt( .299 R2 + .587 G2 + .114 B2 ), see http://alienryderflex.com/hsp.html
    // HSP rule sqrt( .299 R2 + .587 G2 + .114 B2 ), see http://alienryderflex.com/hsp.html
    VerticalTimelineCardComponent.prototype.calculateBrightness = 
    // HSP rule sqrt( .299 R2 + .587 G2 + .114 B2 ), see http://alienryderflex.com/hsp.html
    function (R, G, B) {
        return Math.sqrt((0.299 * Math.pow(R, 2)) +
            (0.587 * Math.pow(G, 2)) +
            (0.114 * Math.pow(B, 2)));
    };
    /** @nocollapse */
    VerticalTimelineCardComponent.propDecorators = {
        "dateValue": [{ type: Input },],
        "timeString": [{ type: Input },],
        "color": [{ type: Input },],
        "disabled": [{ type: Input },],
        "isATimelineItem": [{ type: HostBinding, args: ['class.timeline-item',] },],
    };
VerticalTimelineCardComponent.ɵfac = function VerticalTimelineCardComponent_Factory(t) { return new (t || VerticalTimelineCardComponent)(); };
VerticalTimelineCardComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: VerticalTimelineCardComponent, selectors: [["vertical-timeline-card"]], hostVars: 2, hostBindings: function VerticalTimelineCardComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("timeline-item", ctx.isATimelineItem);
    } }, inputs: { disabled: "disabled", dateValue: "dateValue", color: "color", timeString: "timeString" }, ngContentSelectors: _c1, decls: 11, vars: 15, consts: [[1, "timeline-img"], [3, "color", 4, "ngIf"], [1, "timeline-content", 3, "ngClass"], [1, "date"], [1, "inner-content"]], template: function VerticalTimelineCardComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵtemplate(2, VerticalTimelineCardComponent_p_2_Template, 3, 6, "p", 1);
        ɵngcc0.ɵɵtemplate(3, VerticalTimelineCardComponent_p_3_Template, 2, 3, "p", 1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(4, "div", 2);
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelementStart(6, "p");
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵpipe(8, "date");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 4);
        ɵngcc0.ɵɵprojection(10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.timeString || ctx.timeString === "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.timeString && ctx.timeString != "");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(13, _c0, ctx.disabled));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("color", ctx.textColor);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate(ɵngcc0.ɵɵpipeBind2(8, 10, ctx.dateValue, "dd MMMM yyyy"));
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass], pipes: [ɵngcc1.DatePipe], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(VerticalTimelineCardComponent, [{
        type: Component,
        args: [{
                selector: 'vertical-timeline-card',
                template: "\n      <div>\n        <div class=\"timeline-img\" [style.background]=\"color\">\n          <p *ngIf=\"!timeString || timeString === ''\" [style.color]=\"textColor\">{{dateValue | date: \"HH:mm\"}}</p>\n          <p *ngIf=\"timeString && timeString != ''\" [style.color]=\"textColor\">{{timeString}}</p>\n        </div>\n\n        <div class=\"timeline-content\" [ngClass]=\"{'disabled-element': disabled}\">\n          <div class=\"date\" [style.background]=\"color\">\n            <p [style.color]=\"textColor\">{{dateValue | date:\"dd MMMM yyyy\"}}</p>\n          </div>\n\n          <div class=\"inner-content\">\n            <ng-content></ng-content>\n          </div>\n        </div>\n      </div>\n    "
            }]
    }], function () { return []; }, { disabled: [{
            type: Input
        }], isATimelineItem: [{
            type: HostBinding,
            args: ['class.timeline-item']
        }], dateValue: [{
            type: Input
        }], color: [{
            type: Input
        }], timeString: [{
            type: Input
        }] }); })();
    return VerticalTimelineCardComponent;
}());
export { VerticalTimelineCardComponent };
