// ngx-horizontal-timeline/__ivy_ngcc__/fesm2015/ngx-horizontal-timeline.js
import { Injectable, Component, Input, NgModule, defineInjectable } from '@angular/core';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function NgxHorizontalTimelineComponent_span_2_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r1.label);
} }
function NgxHorizontalTimelineComponent_span_2_Template(rf, ctx) { if (rf & 1) {
    const _r5 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtemplate(1, NgxHorizontalTimelineComponent_span_2_div_1_Template, 2, 1, "div", 4);
    ɵngcc0.ɵɵelementStart(2, "div", 5);
    ɵngcc0.ɵɵlistener("click", function NgxHorizontalTimelineComponent_span_2_Template_div_click_2_listener() { const restoredCtx = ɵngcc0.ɵɵrestoreView(_r5); const item_r1 = restoredCtx.$implicit; const ctx_r4 = ɵngcc0.ɵɵnextContext(); return ctx_r4.activeItem(item_r1); });
    ɵngcc0.ɵɵelement(3, "i");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r1 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(item_r1.styleClass ? item_r1.styleClass : "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", item_r1.label);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(item_r1.active ? "marker timeline-icon marker-active" : "marker timeline-icon");
    ɵngcc0.ɵɵstyleProp("background-color", ctx_r0.getColor(item_r1));
    ɵngcc0.ɵɵpropertyInterpolate("title", item_r1.title);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(item_r1.icon);
} }
class NgxHorizontalTimelineService {
    constructor() { }
}
NgxHorizontalTimelineService.ɵfac = function NgxHorizontalTimelineService_Factory(t) { return new (t || NgxHorizontalTimelineService)(); };
NgxHorizontalTimelineService.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: NgxHorizontalTimelineService, factory: NgxHorizontalTimelineService.ɵfac, providedIn: 'root' });
/** @nocollapse */
NgxHorizontalTimelineService.ctorParameters = () => [];
/** @nocollapse */ NgxHorizontalTimelineService.ngInjectableDef = defineInjectable({ factory: function NgxHorizontalTimelineService_Factory() { return new NgxHorizontalTimelineService(); }, token: NgxHorizontalTimelineService, providedIn: "root" });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxHorizontalTimelineService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgxHorizontalTimelineComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() { }
    /**
     * @param {?} item
     * @return {?}
     */
    activeItem(item) {
        for (const i of this.items) {
            i.active = undefined;
        }
        item.active = true;
        if (item.command) {
            item.command();
        }
    }
    /**
     * @param {?} item
     * @return {?}
     */
    getColor(item) {
        if (!item.color) {
            return null;
        }
        if (item.active) {
            return '#' + item.color;
        }
        else {
            return '#' + this.lighten(item.color, 50);
        }
    }
    /**
     * @param {?} col
     * @param {?} amt
     * @return {?}
     */
    lighten(col, amt) {
        col = parseInt(col, 16);
        return (
        // tslint:disable-next-line:no-bitwise
        ((col & 0x0000ff) + amt) | ((((col >> 8) & 0x00ff) + amt) << 8) | (((col >> 16) + amt) << 16)).toString(16);
    }
}
NgxHorizontalTimelineComponent.ɵfac = function NgxHorizontalTimelineComponent_Factory(t) { return new (t || NgxHorizontalTimelineComponent)(); };
NgxHorizontalTimelineComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: NgxHorizontalTimelineComponent, selectors: [["ngx-htl-horizontal-timeline"]], inputs: { items: "items" }, decls: 4, vars: 1, consts: [["id", "timeline-wrap"], [1, "navigation"], [3, "class", 4, "ngFor", "ngForOf"], [1, "timeline"], ["class", "timeline-label", 4, "ngIf"], [3, "title", "click"], [1, "timeline-label"]], template: function NgxHorizontalTimelineComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtemplate(2, NgxHorizontalTimelineComponent_span_2_Template, 4, 10, "span", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc1.NgForOf, ɵngcc1.NgIf], styles: [".navigation[_ngcontent-%COMP%]{display:flex;flex-flow:row wrap;justify-content:space-around;align-items:center}.timeline[_ngcontent-%COMP%]{height:1px;width:100%;position:absolute;background:#aabbc4}.timeline[_ngcontent-%COMP%]:before{background:0 0}.timeline-label[_ngcontent-%COMP%]{text-align:center;position:absolute;width:50px;bottom:55px;color:#555;font-family:\"Source Sans Pro\",\"Helvetica Neue\",Helvetica,Arial,sans-serif!important;font-weight:400}@media all and (max-width:500px){.navigation[_ngcontent-%COMP%]{flex-direction:row;justify-content:initial}.timeline[_ngcontent-%COMP%]{height:0}.marker[_ngcontent-%COMP%]{margin-top:25px}.timeline-title[_ngcontent-%COMP%]{position:relative;bottom:-14px}}#timeline-wrap[_ngcontent-%COMP%]{margin:4% 5%;position:relative}.marker[_ngcontent-%COMP%]{z-index:1;color:#fff;width:50px;height:50px;line-height:50px;font-size:1.4em;text-align:center;position:relative;background-color:#999;border-radius:50%}.marker[_ngcontent-%COMP%]:hover{-webkit-transform:scale(1.3);transform:scale(1.3);cursor:pointer;transition:.3s}.timeline-icon[_ngcontent-%COMP%]{background-color:#7e8aaf}.marker-active[_ngcontent-%COMP%]{-webkit-transform:scale(1.3);transform:scale(1.3);background-color:#3e4f88}"] });
/** @nocollapse */
NgxHorizontalTimelineComponent.ctorParameters = () => [];
NgxHorizontalTimelineComponent.propDecorators = {
    items: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxHorizontalTimelineComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-htl-horizontal-timeline',
                template: "<div id=\"timeline-wrap\">\n  <div class=\"navigation\">\n\n    <span *ngFor=\"let item of items\" [class]=\"item.styleClass? item.styleClass: ''\">\n      <div class=\"timeline-label\" *ngIf=\"item.label\">{{item.label}}</div>\n\n      <div (click)=\"activeItem(item)\" [style.background-color]=\"getColor(item)\"\n        title=\"{{item.title}}\"\n        [class]=\"\n          item.active\n            ? 'marker timeline-icon marker-active'\n            : 'marker timeline-icon'\">\n        <i [class]=\"item.icon\"></i>\n      </div>\n    </span>\n    <div class=\"timeline\"></div>\n  </div>\n</div>\n",
                styles: [".navigation{display:flex;flex-flow:row wrap;justify-content:space-around;align-items:center}.timeline{height:1px;width:100%;position:absolute;background:#aabbc4}.timeline:before{background:0 0}.timeline-label{text-align:center;position:absolute;width:50px;bottom:55px;color:#555;font-family:\"Source Sans Pro\",\"Helvetica Neue\",Helvetica,Arial,sans-serif!important;font-weight:400}@media all and (max-width:500px){.navigation{flex-direction:row;justify-content:initial}.timeline{height:0}.marker{margin-top:25px}.timeline-title{position:relative;bottom:-14px}}#timeline-wrap{margin:4% 5%;position:relative}.marker{z-index:1;color:#fff;width:50px;height:50px;line-height:50px;font-size:1.4em;text-align:center;position:relative;background-color:#999;border-radius:50%}.marker:hover{-webkit-transform:scale(1.3);transform:scale(1.3);cursor:pointer;transition:.3s}.timeline-icon{background-color:#7e8aaf}.marker-active{-webkit-transform:scale(1.3);transform:scale(1.3);background-color:#3e4f88}"]
            }]
    }], function () { return []; }, { items: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgxHorizontalTimelineModule {
}
NgxHorizontalTimelineModule.ɵfac = function NgxHorizontalTimelineModule_Factory(t) { return new (t || NgxHorizontalTimelineModule)(); };
NgxHorizontalTimelineModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: NgxHorizontalTimelineModule });
NgxHorizontalTimelineModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxHorizontalTimelineModule, [{
        type: NgModule,
        args: [{
                declarations: [NgxHorizontalTimelineComponent],
                imports: [
                    CommonModule
                ],
                exports: [NgxHorizontalTimelineComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxHorizontalTimelineModule, { declarations: function () { return [NgxHorizontalTimelineComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxHorizontalTimelineComponent]; } }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgxHorizontalTimelineService, NgxHorizontalTimelineComponent, NgxHorizontalTimelineModule };
